### ISSUE TITLE
Repetition on group of alternatives is handled incorrectly

### ISSUE THREAD

**Issue by zmajeed** [zmajeed/ebnfparser#6]
The rule for repetition on groups works for a concatenation group but not for a group of alternatives

For example, this EBNF generates wrong BNF

```
<separator> ::=
    { <comment>
  | <whitespace> }...
```

**Comment by zmajeed** [comment]
The right way is to first replace the group with a new nonterminal like `comment_or_whitespace`, then replace the repetition on `comment_or_whitespace` with `comment_or_whitespace_list`

The result would be

```
separator: comment_or_whitespace_list

comment_or_whitespace_list: comment_or_whitespace | comment_or_whitespace_list comment_or_whitespace

comment_or_whitespace: comment | whitespace
```

This solution actually works for concatenation groups too - so one fix is to replace a group with a new nonterminal like `group_0` regardless

```
group: "{" production_combos "}" {
// groups are replaced by new single nonterminal
// move all productions of group to new rule for new nonterminal
  stringstream groupName;
  groupName << "group_" << groupNumber;
  ++groupNumber;
  bisonParam.result[groupName.str()] = $production_combos;
  $$.insert({groupName.str()});
}

repetition: group "..." {
  auto groupName = (*$group.begin())[0];
  auto listRuleName = groupName + "_list";
  bisonParam.result[listRuleName] = { {groupName}, {listRuleName, groupName} };
  $$ = { {listRuleName} };
}
```

This isn't great because it creates many new nonterminals and rules - concatenation groups are common in the GQL grammar - but only a few groups of alternatives - and just one repetition on a group of alternatives

It would be nice if we can simply restrict the current rule for repetition on group to concatenation groups - and add a new rule for repetition on a group of alternatives

But it's not as simple as something like

```
repetition: "{" concatenation "}" "..." {
}
| "{" alternative "}" "..." {
}
```

because this creates shift-reduce conflicts with the rule for `production_combos`

```
production_combos: concatenation | alternative
```

And `production_combos` is probably impossible to remove or change because it's crucial to the recursive rules in the grammar

I think the better fix is to tag a group as alternative or concatenation and use the tag to generate different BNF in the action for repetition - something like

```
struct Group {
  enum class Type {concatenation, alternative};
  Production productions;
};
```

**Comment by zmajeed** [comment]
Fixed with #11
